### ISSUE TITLE
Composable table - remove hasSelectableRowCaption prop

### ISSUE THREAD

**Issue by wise-king-sullyman** [patternfly/patternfly-react#7473]
Followup to #7447

On the next breaking change release, the above mentioned prop should be removed to facilitate better automatic accessibility.

**Comment by nicolethoen** [comment]
Closed by #8352
